bazel/llvm_project/0004_Introduce_filegroups_for_compiler_rt_builtins_runtime.patch

```diff
@@ -0,0 +1,193 @@
++    srcs = glob(BUILTINS_BF16_SRCS_PATTERNS),
++)
++
++BUILTINS_X86_FP80_SRCS_PATTERNS = [
++    # `xc` marks 80-bit complex number builtins.
++    "lib/builtins/*xc*.c",
++
++    # `xf` marks 80-bit floating-point builtins.
++    "lib/builtins/*xf*.c",
++]
++
++filegroup(
++    name = "builtins_x86_fp80_srcs",
++    srcs = glob(
++        BUILTINS_X86_FP80_SRCS_PATTERNS,
++        exclude = BUILTINS_BF16_SRCS_PATTERNS,
++    ),
++)
++
++BUILTINS_TF_SRCS_PATTERNS = [
++    # `tc` marks 128-bit complex number builtins.
++    "lib/builtins/*tc*.c",
++
++    # `tf` marks 128-bit floating-point builtins.
++    "lib/builtins/*tf*.c",
++]
++
++BUILTINS_TF_EXCLUDES = BUILTINS_HOSTED_SRCS + BUILTINS_BF16_SRCS_PATTERNS + BUILTINS_X86_FP80_SRCS_PATTERNS
++
++filegroup(
++    name = "builtins_tf_srcs",
++    srcs = glob(
++        BUILTINS_TF_SRCS_PATTERNS,
++        exclude = BUILTINS_TF_EXCLUDES,
++    ),
++)
++
++BUILTINS_MACOS_ATOMIC_SRCS_PATTERNS = [
++    "lib/builtins/atomic_*.c",
++]
++
++filegroup(
++    name = "builtins_macos_atomic_srcs",
++    srcs = glob(BUILTINS_MACOS_ATOMIC_SRCS_PATTERNS),
++)
++
++filegroup(
++    name = "builtins_aarch64_srcs",
++    srcs = [
++        "lib/builtins/cpu_model/aarch64.c",
++        "lib/builtins/cpu_model/aarch64.h",
++    ] + glob(
++        [
++            "lib/builtins/cpu_model/AArch64*.inc",
++            "lib/builtins/cpu_model/aarch64/**/*.inc",
++            "lib/builtins/aarch64/*.S",
++            "lib/builtins/aarch64/*.c",
++        ],
++        exclude = [
++            # This file isn't intended to directly compile, but to be used to
++            # generate a collection of outline atomic helpers.
++            # TODO: Add support for generating the sources for these helpers if
++            # there are users that need this functionality from the builtins
++            # library.
++            "lib/builtins/aarch64/lse.S",
++        ],
++    ),
++)
++
++filegroup(
++    name = "builtins_x86_arch_srcs",
++    srcs = [
++        "lib/builtins/cpu_model/x86.c",
++        "lib/builtins/i386/fp_mode.c",
++    ],
++)
++
++filegroup(
++    name = "builtins_x86_64_srcs",
++    srcs = glob([
++        "lib/builtins/x86_64/*.c",
++        "lib/builtins/x86_64/*.S",
++    ]),
++)
++
++filegroup(
++    name = "builtins_i386_srcs",
++    srcs = glob(
++        [
++            "lib/builtins/i386/*.c",
++            "lib/builtins/i386/*.S",
++        ],
++        exclude = [
++            # This file is used for both i386 and x86_64.
++            "lib/builtins/i386/fp_mode.c",
++        ],
++    ),
++)
++
++filegroup(
++    name = "builtins_generic_srcs",
++    srcs = ["lib/builtins/cpu_model/cpu_model.h"] + glob(
++        [
++            "lib/builtins/*.c",
++            "lib/builtins/*.h",
++            "lib/builtins/*.inc",
++        ],
++        exclude = BUILTINS_CRTBEGIN_SRCS + BUILTINS_CRTEND_SRCS + BUILTINS_TF_EXCLUDES + BUILTINS_TF_SRCS_PATTERNS + BUILTINS_MACOS_ATOMIC_SRCS_PATTERNS,
```

Comment:
Minor, and don't bother if you don't want to regenerate the patch, but here and line 108, you might be able to add parens in order to wrap the additions (but I don't know how buildifier would react). i.e.:

```
exclude = (
  x +
  y +
  z
),
```

Reply:
Done since I was regenerating anyways.